src/main/java/io/nats/client/impl/NatsMessage.java

```diff
@@ -133,24 +107,83 @@ static int copy(byte[] dest, int pos, String toCopy) {
         if (replyTo != null) {
             this.replyTo = replyTo;
         }
-        this.sizeInBytes = protocolLength + 2;
+        this.protocolLength = protocolLength;
         this.data = null; // will set data and size after we read it
     }
 
+    private static int fastIntLength(int number) {

```

Comment:
This implementation while a bit verbose is significantly faster than a naive implementation and allows us to eliminate a separate length buffer which lets us write directly to `protocolBytes` avoiding some somewhat expensive copy operations.